src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+    private int capacity;
```

Comment:
it's quite useless as a field, we can call .length on elements when we need it, not worth to keep it

Reply:
After removing the variable, it's not completely clear how to increment the size array.

Reply:
In method resize you should calculate new capacity `int newCapacity = (int) (elements.length * CAPACITY_INDEX);` and create newArray with this newCapacity, then just copy old array into new array

Reply:
> it's quite useless as a field, we can call .length on elements when we need it, not worth to keep it

I mean, that in different code parts the field (the array size) changes. For example, it increases: 
       `resize();
        System.arraycopy(elements, index, elements, index + 1, size - index);
        elements[index] = value;
        size++;`
decreases:
`public T remove(int index) {
        checkIndex(index);
        T arrayElement = elements[index];
        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
        elements[--size] = null;
        return arrayElement;
    }`.
.length is fixed.

> In method resize you should calculate new capacity `int newCapacity = (int) (elements.length * CAPACITY_INDEX);` and create newArray with this newCapacity, then just copy old array into new array

Also, in each method, this field is used differently. Need to create multiple methods (create and copy new arrays) and is that better than one field?

Reply:
Let's discussed it in chat on the platform?

Reply:
> it's quite useless as a field, we can call .length on elements when we need it, not worth to keep it

Sorry, I thought that the useless field was the 'size' field.